Add a tutorial

ChrisRackauckas opened the pull request:
I am a firm believer that for any software library, a tutorial needs to come first before the manual to "show before telling" and introduce the terminology of the library in a more natural way. Plots.jl has been missing a tutorial. This is that tutorial.

In this proposed version, I walk through the basics to show how to generate useful plots and touch on "common gotchas". The goal is simply to get the user in a spot where they can easily understand the documentation to get more details, and not to cover everything.

Currently I have not added the actual plots in case things need to be changed around. But if this passes review, I plan to add the plot from each plot command directly underneath the command, making it a very visual introduction to Plots.jl.

Calling in some reviewers.  @mkborregaard and @daschw as masters of Plots.jl. @dpsanders and @albi3ro as masters of Julia tutorials and introductions. Anyone else feel free to join in as well of course!

dpsanders commented:
+1000 for the idea and the effort! Won't have a chance to check the details for a while. Please remind me in a week.

mkborregaard commented:
What @dpsanders  said! (on both accounts unfortunately, except that you needn't remind me :-) )

tbreloff commented:
+100. Great stuff Chris!

On Fri, Jul 28, 2017 at 11:20 AM Michael Krabbe Borregaard <
notifications@github.com> wrote:

> What @dpsanders <https://github.com/dpsanders> said! (on both accounts
> unfortunately, except that you needn't remind me :-) )
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaPlots/PlotDocs.jl/pull/38#issuecomment-318681592>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA492i1htJpjZAlCgQdjUVbxY9-QEN1Iks5sSfxAgaJpZM4Omtat>
> .
>

daschw commented on docs/tutorial.md:
````diff
@@ -0,0 +1,284 @@
+
+```julia
+x = 1:10; y = rand(10,2) # 2 columns means two lines
+plot(x,y,title="Two Lines",label=["Line 1","Line 2"],lw=3)
````
columns are series, right? ;-) `label=["Line 1" "Line 2"]`

mkborregaard commented on docs/tutorial.md:
```diff
@@ -0,0 +1,284 @@
+
+## Plot Attributes
+
+In the previous section we made plots... we're done right? No! We need to style
```
I think it would be nice to mention what I think of as a basic premise for plot calls - that positional arguments to plot are the input data to be plotted, and keyword arguments are attributes determining how they are plotted (as an aside this is actually the underlying premise of the grammar of graphics - I just think it is much more elegantly implemented in Plots, but that's no secret).

mkborregaard commented on docs/tutorial.md:
````diff
@@ -0,0 +1,284 @@
+plot(x,y,title="Two Lines",label=["Line 1" "Line 2"],lw=3)
+```
+
+Note that every attribute also has a modifier function for it. For example, the
````
"Note that every attribute can also be applied by mutating the plot with a modifier function" would be slightly more clear, I think.

mkborregaard commented on docs/tutorial.md:
```diff
@@ -0,0 +1,284 @@
+
+When we started plotting above, our plot used the default backend. The default
+backend depends on what plotting packages you've installed in Julia. If you're
+using a basic installation, this will plot to the browser using Plotly, or into
```
I'd say "If you've got no backend packages installed"

mkborregaard commented on docs/tutorial.md:
```diff
@@ -0,0 +1,284 @@
+[StatPlots.jl](https://github.com/JuliaPlots/StatPlots.jl). Let's look into
+StatPlots.jl. StatPlots.jl does two things:
+
+1) It adds a type recipe for `DataFrame`s.
```
It actually doesn't - it is a user recipe (an unfortunate effect of this is that only series recipes are compatible with DataFrames out of the box)

mkborregaard commented on docs/tutorial.md:
```diff
@@ -0,0 +1,284 @@
+2) It adds a bunch of statistical plot series.
+
+### Using Type Recipes
+
```
Again, in the Plots terminology, this is called user recipes. Type recipes are drop-in replacements for vectors.

mkborregaard replied:
After thinking about this, I wouldn't be against simply changing the Plots terminology to what you use here - it'd be a bit involved, though, as the terms are used internally.

ChrisRackauckas replied:
I was wrong with the type recipe because yeah, this is more than just a data transformation since the `plot` command is changed to symbols, but the name "user recipe" has always been non-descriptive to me.

mkborregaard replied:
Yes, exactly. User recipe could be changed to type recipe, and type recipe to "data recipe", "argument recipe" or perhaps "translation recipe".

mkborregaard replied:
I'm not sure it's important, though, it's a quite big change

mkborregaard commented:
@PaulSoderlind I think you requested something like this?

PaulSoderlind commented:
Very useful

ChrisRackauckas commented:
Updated to address @mkborregaard 's commands and added a plot for each plot command. Please take a quick second look to see if this is to your liking.

mkborregaard commented:
Nice - I noticed on some of the GR plots the title is behind the plot - is your Plots master fully up-to-date?

ChrisRackauckas commented:
Indeed I was using an old version. Nice to see this fixed. I updated the GR plots.

ChrisRackauckas commented:
>feel free to merge this yourself when you're happy with it!

Can I get the go ahead?

mkborregaard commented:
Hi @ChrisRackauckas thanks for asking and sorry for being unclear. You are 100% good to go, thanks for this!
